Community/Tdarr_Plugin_advo_FFMPEG_NVENC_Tiered_MKV_with_rollback.js

```diff
@@ -0,0 +1,563 @@
+      + '-metadata ADVOVTRANSCODEDISCARDED=true ';
+    return response;
+  }
+
+  // find video stream
+  for (let i = 0; i < file.ffProbeData.streams.length; i += 1) {
+    const currStream = file.ffProbeData.streams[i];
+
+    try {
+      if (currStream.codec_type.toLowerCase() === 'video') {
+        if (currStream.codec_name.toLowerCase() !== 'png'
+          && currStream.codec_name.toLowerCase() !== 'bmp'
+          && currStream.codec_name.toLowerCase() !== 'gif'
+          && currStream.codec_name.toLowerCase() !== 'mjpeg') {
+          videoTrackIdx = i;
+          videoStream = currStream;
+          break;
+        }
+      }
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.log(err);
+    }
+  }
+
+  // find video media info track
+  for (let i = 1; i < file.mediaInfo.track.length; i += 1) {
+    const currTrack = file.mediaInfo.track[i];
+
+    try {
+      if (currTrack.StreamOrder !== undefined && parseInt(currTrack.StreamOrder, 10) === videoTrackIdx) {
+        videoTrack = currTrack;
+      }
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.log(err);
+    }
+  }
+
+  if (inputs.bframe > 0) {
+    // only use temporal_aq if bframes are enabled.
+    extraOptions += ` -bf ${inputs.bframe} -b_ref_mode 1 -temporal_aq:v 1 -aq-strength:v 8 `;
+  }
+
+  if (typeof file.meta.Duration !== 'undefined') {
+    duration = file.meta.Duration / 60;
+  } else {
+    duration = videoTrack.Duration / 60;
+  }
+
+  // check if the file is a video, if not the function will be stopped immediately
+  if (file.fileMedium !== 'video') {
+    response.processFile = false;
+    response.infoLog += '☒File is not a video! \n';
+    return response;
+  }
+
+  bitrateprobe = videoTrack.BitRate;
+  response.infoLog += `☑ BitRate: ${bitrateprobe} \n`;
+
+  if (bitrateprobe == null || bitrateprobe < 100000) {
+    bitrateprobe = videoTrack.OverallBitRate;
+    response.infoLog += `☑ OverallBitRate: ${bitrateprobe} \n`;
+
+    if (bitrateprobe == null || bitrateprobe < 100000) {
+      bitrateprobe = videoTrack.BitRate_Maximum;
+      response.infoLog += `☑ BitRate_Maximum: ${bitrateprobe} \n`;
+
+      if (bitrateprobe == null || bitrateprobe < 100000) {
+        bitrateprobe = file.bit_rate;
+        response.infoLog += `☑ bit_rate: ${bitrateprobe} \n`;
+
+        if (bitrateprobe == null || bitrateprobe < 100000) {
+          bitrateprobe = (file.file_size / (duration * 0.0075)) * 1024;
+          response.infoLog += `☑ estimated bit rate: ${bitrateprobe} \n`;
+        }
+      }
+    }
+  }
+
+  response.infoLog += '☑File is a video! \n';
+
+  if (file.mediaInfo.track[0].extra !== undefined
+    && file.mediaInfo.track[0].extra.ADVOVTRANSCODEDONE !== undefined
+    && file.mediaInfo.track[0].extra.ADVOVTRANSCODEDONE === 'true') {
+    response.processFile = false;
+    response.infoLog += '☑File has already been transcoded! \n';
+    return response;
+  }
```

Comment:
Why does it need two checks like this? The one on line 222 does *roughly* the same.

Reply:
I can rework the if on 222 and combine, but it's a different purpose. 222 is checking if the new file is larger than the old file, and reverting. This section is just seeing if the file has previously been encoded already.